### PR TITLE
Enable App Sandbox.

### PR DESCRIPTION
**Pull request by nevack**

### CODE REVIEW

File: Application/PlaybackEventController.m [hunk 1]
```diff
@@ -12,8 +12,6 @@
 
 #import "AudioScrobbler.h"
 
-NSString *TrackNotification = @"com.apple.iTunes.playerInfo";
```
Comment (nevack) [inline Application/PlaybackEventController.m:15]: Got sandbox warnings for doing so.

**Comment by nevack** [comment]
The only error I see now on startup
```
2021-02-26 16:17:07.363432+0300 Cog[28758:137078] Error loading /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay:  dlopen(/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay, 265): Library not loaded: @rpath/sidplayfp.framework/Versions/A/sidplayfp
  Referenced from: /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay
  Reason: no suitable image found.  Did find:
	file system sandbox blocked open() of '/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/sidplayfp.framework/Versions/A/sidplayfp'
```

**Comment by nevack** [comment]
> The only error I see now on startup
> 
> ```
> 2021-02-26 16:17:07.363432+0300 Cog[28758:137078] Error loading /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay:  dlopen(/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay, 265): Library not loaded: @rpath/sidplayfp.framework/Versions/A/sidplayfp
>   Referenced from: /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay
>   Reason: no suitable image found.  Did find:
> 	file system sandbox blocked open() of '/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/sidplayfp.framework/Versions/A/sidplayfp'
> ```

@kode54 Quick googling shows that the error like this is only reproducible with free Apple Developer accounts. May I ask you to test build from this PR on your machine?

**Comment by nevack** [comment]
> > The only error I see now on startup
> > ```
> > 2021-02-26 16:17:07.363432+0300 Cog[28758:137078] Error loading /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay:  dlopen(/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay, 265): Library not loaded: @rpath/sidplayfp.framework/Versions/A/sidplayfp
> >   Referenced from: /Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/Cog.app/Contents/PlugIns/sidplay.bundle/Contents/MacOS/sidplay
> >   Reason: no suitable image found.  Did find:
> > 	file system sandbox blocked open() of '/Users/nevack/Library/Developer/Xcode/DerivedData/Cog-ffujrjjlvwctciftlnbjtlxqzdql/Build/Products/Debug/sidplayfp.framework/Versions/A/sidplayfp'
> > ```
> 
> @kode54 Quick googling shows that the error like this is only reproducible with free Apple Developer accounts. May I ask you to test build from this PR on your machine?

Added framework as dep to Cog target and the issue is gone.

File: Cog.entitlements [hunk 2]
```diff
@@ -2,9 +2,17 @@
 <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <dict>
+	<key>com.apple.security.app-sandbox</key>
```
Comment (nevack) [inline Cog.entitlements:5]: Permissions requested:
1. Incoming Connections (http stream?)
2. User's Music directory read-only access (play user's music)
3. Drag&droped and opened files read-only access (play user's music)

I think this is sufficient for now.

**Comment by nevack** [comment]
Rebase + Conflict fix

File: Cog.entitlements (hunk 2 above)
Comment (kode54) [reply Cog.entitlements:5]: Will also need to store bookmarks for those dragged and dropped or opened music files, unless the XML playlist is already sufficient for that.

**Comment by kode54** [comment]
I will make another commit to this branch that fixes libsidplayfp without moving it into the main app, by changing the install base path to match all other subproject Frameworks.

This doesn't fix Sandbox accessing files outside of the Music folder across executions. We need to resolve those using bookmarks or something.

**Comment by kode54** [comment]
Hell with it, I don't know how to edit pull requests off the site, so I just merged it.

**Pull request merged**